pkg/azure/ocpgwdeployer.go

```diff
@@ -267,16 +267,35 @@ func (d *ocpGatewayDeployer) initMachineSet(name, zone string) (*unstructured.Un
 }
 
 func (d *ocpGatewayDeployer) deployGateway(zone string) error {
-	name := d.azure.InfraID + submarinerGatewayGW + d.azure.Region + "-" + zone
-
-	machineSet, err := d.initMachineSet(name, zone)
+	machineSet, err := d.initMachineSet(MachineName(d.azure.InfraID, d.azure.Region, zone), zone)
 	if err != nil {
 		return err
 	}
 
 	return errors.Wrapf(d.msDeployer.Deploy(machineSet), "error deploying machine set %q", machineSet.GetName())
 }
 
+func MachineName(infraID, region, zone string) string {
+	// Limit the name length to 40 characters to ensure we don't hit the 63-character limit
+	// when generating the "machine public IP name".
+	// At most 6 characters for the zone, at most 12 for the region and zone combined,
+	// at most 32 for the infra id, region and zone combined.
+	// We add "-subgw-", 7 characters, for a total of 40 with the hyphen between region and zone.
```

Comment:
Wouldn't it be better as a function doc?

Reply:
Why do we specifically limit the zone to 6 and the region to 12?